thirdparty/Annoy/CMakeLists.txt

```diff
@@ -0,0 +1,40 @@
+# This file is part of the Acts project.
+#
+# Copyright (C) 2024 CERN for the benefit of the Acts project
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# CMake include(s).
+include(FetchContent)
+
+# Minimum CMake version.
+cmake_minimum_required(VERSION 3.14) # FetchContent_MakeAvailable was introduced in 3.14
+
+# Information message.
+message(STATUS "Building Annoy as part of the ACTS project")
+
+# Version of the Annoy library to fetch.
+set(ANNOY_VERSION "v${_acts_annoy_version}")
+
+# Git repository for the Annoy library.
+set(ACTS_ANNOY_GIT_REPOSITORY "https://github.com/spotify/annoy.git"
+   CACHE STRING "Git repository to take Annoy from")
+
+# Declare the fetching of the Annoy library.
+FetchContent_Declare(
+   Annoy
+   GIT_REPOSITORY "${ACTS_ANNOY_GIT_REPOSITORY}"
+   GIT_TAG "${ANNOY_VERSION}"
+   CONFIGURE_COMMAND ""
+   BUILD_COMMAND ""
+   INSTALL_COMMAND ""
+   PATCH_COMMAND "${CMAKE_COMMAND}" -E chdir <SOURCE_DIR> ${GIT_EXECUTABLE} config user.name "Jeremy Couthures" &&
+                  "${CMAKE_COMMAND}" -E chdir <SOURCE_DIR> ${GIT_EXECUTABLE} config user.email "jeremy.couthures@lapp.in2p3.fr" &&
```

Comment:
I think name and email can be dropped

Reply:
I get the following error in the building configuration of the CI when I drop them:

```
Cloning into 'annoy-src'...
HEAD is now at 75429e5 Version 1.17.3
[2/9] Performing update step for 'annoy-populate'
[4/9] Performing patch step for 'annoy-populate'
-- Configuring incomplete, errors occurred!
See also "/__w/acts/acts/build/CMakeFiles/CMakeOutput.log".
FAILED: annoy-populate-prefix/src/annoy-populate-stamp/annoy-populate-patch /__w/acts/acts/build/_deps/annoy-subbuild/annoy-populate-prefix/src/annoy-populate-stamp/annoy-populate-patch 
cd /__w/acts/acts/build/_deps/annoy-src && /usr/bin/cmake -E chdir /__w/acts/acts/build/_deps/annoy-src /usr/bin/git am /__w/acts/acts/thirdparty/Annoy/0001-Modify-annoy.patch && /usr/bin/cmake -E touch /__w/acts/acts/build/_deps/annoy-subbuild/annoy-populate-prefix/src/annoy-populate-stamp/annoy-populate-patch
Committer identity unknown

*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: unable to auto-detect email address (got 'root@5b8552de0b35.(none)')
ninja: build stopped: subcommand failed.

CMake Error at /usr/share/cmake-3.22/Modules/FetchContent.cmake:1087 (message):
  Build step for annoy failed: 1
Call Stack (most recent call first):
  /usr/share/cmake-3.22/Modules/FetchContent.cmake:1216:EVAL:2 (__FetchContent_directPopulate)
  /usr/share/cmake-3.22/Modules/FetchContent.cmake:1216 (cmake_language)
  /usr/share/cmake-3.22/Modules/FetchContent.cmake:1259 (FetchContent_Populate)
  thirdparty/Annoy/CMakeLists.txt:38 (FetchContent_MakeAvailable)
```

Reply:
we have a similar fix here https://github.com/acts-project/acts/blob/5ca5ec0bbfcbf98afae83a8a7d29b08cac7834db/.github/workflows/builds.yml#L55-L56

Reply:
I only have the issue with the ubuntu2204 builds. I see that the fix you mentioned is not used for those builds. Is it intended?
https://github.com/acts-project/acts/blob/5ca5ec0bbfcbf98afae83a8a7d29b08cac7834db/.github/workflows/builds.yml#L264

Reply:
having it in the cmake is fine I would just not put an actual name or an email address there

Reply:
Oh, I understand, I will change them to 'CI' and '<>' to match the workflow.